.github/workflows/ci.yml

```diff
@@ -23,24 +21,21 @@ jobs:
     steps:
     - name: Checkout
       uses: actions/checkout@v3
-    - name: Setup Python 3.8
+    - name: Setup Python
       uses: actions/setup-python@v4
       with:
         python-version: 3.8
-    - name: Cache PyPI
-      uses: actions/cache@v3
-      with:
-        key: pip-lint-${{ hashFiles('requirements.txt') }}
-        path: ~/.cache/pip
-        restore-keys: |
-            pip-lint-
+        cache: 'pip'
+        cache-dependency-path: '**/requirements*.txt'
+    - name: Pre-Commit hooks
+      uses: pre-commit/action@v3.0.0
     - name: Install dependencies
       uses: py-actions/py-dependency-install@v4.0.0
       with:
-        path: requirements.txt
+        path: requirements-dev.txt
     - name: Install itself
       run: |
-        python setup.py install
+        pip install .
     - name: Run linter
       run: |
         make lint
```

Comment:
@Dreamsorcerer I've been experimenting with modularizing the workflows via reuse. See https://github.com/aio-libs/aiomonitor/blob/9ec9910/.github/workflows/ci-cd.yml#L55-L56 — I'd like to start doing this in other places too.